rpc/types/events.go

```diff
@@ -0,0 +1,210 @@
+package types
+
+import (
+	"encoding/json"
+	"strconv"
+
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+	evmtypes "github.com/tharsis/ethermint/x/evm/types"
+)
+
+// EventFormat is the format version of the events
+type EventFormat int
+
+const (
+	eventFormatUnknown EventFormat = iota
+	eventFormat1
+	eventFormat2
+)
+
+// ParsedTx is the tx infos parsed from events.
+type ParsedTx struct {
+	MsgIndex int
+
+	// the following fields are parsed from events
+
+	// -1 means uninitialized
+	Hash       common.Hash
+	EthTxIndex int64
+	GasUsed    uint64
+	Failed     bool
+	// unparsed tx log json strings
+	RawLogs [][]byte
+}
+
+// NewParsedTx initialize a ParsedTx
+func NewParsedTx(msgIndex int) ParsedTx {
+	return ParsedTx{MsgIndex: msgIndex, EthTxIndex: -1}
+}
+
+// ParseTxLogs decode the raw logs into ethereum format.
+func (p ParsedTx) ParseTxLogs() ([]*ethtypes.Log, error) {
+	logs := make([]*evmtypes.Log, 0, len(p.RawLogs))
+	for _, raw := range p.RawLogs {
+		var log evmtypes.Log
+		if err := json.Unmarshal(raw, &log); err != nil {
+			return nil, err
+		}
+
+		logs = append(logs, &log)
+	}
+	return evmtypes.LogsToEthereum(logs), nil
+}
+
+// ParsedEvents is the tx infos parsed from eth tx events.
+type ParsedTxs struct {
+	// one item per message
+	txs []ParsedTx
+	// map tx hash to msg index
+	txHashes map[common.Hash]int
```

Comment:
q: why make these private?

Reply:
no particular reason, just changed to public.